plugin/plugin.go

```diff
@@ -258,8 +272,11 @@ func (p *AzurePlugin) ListHosts(ctx context.Context, req *pb.ListHostsRequest) (
 				if val.ID == nil { // something went wrong, likely iterator has advanced beyond the end
 					continue
 				}
-				if val.Type == nil || !strings.EqualFold(*val.Type, "Microsoft.Compute/virtualMachines") {
-					// no point continuing if we can't validate that it's a VM
+				if val.Type == nil { // no point continuing if we can't validate that it's a VM
+					continue
+				}
+				if val.Type != nil && (!strings.EqualFold(*val.Type, "Microsoft.Compute/virtualMachines") && !strings.EqualFold(*val.Type, "Microsoft.Compute/virtualMachineScaleSets")) {
```

Comment:
Nit: one set of brackets seem unnecessary, but also, this statement is very long, maybe break it over multiple lines?
```suggestion
				isVm := strings.EqualFold(*val.Type, "Microsoft.Compute/virtualMachines")
				isVmss := strings.EqualFold(*val.Type, "Microsoft.Compute/virtualMachineScaleSets")
				if val.Type != nil && !isVm && !isVmss {
```